### PR TITLE
ISPN-7594 Extract memcached and websocket from the default configs in…

### PR DESCRIPTION
**Pull request by tristantarrant**
…to dedicated config

https://issues.jboss.org/browse/ISPN-7594

### CODE REVIEW

**Comment by tristantarrant** [comment]
@mgencur I'd like your opinion on the changes

File: server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml [hunk 1]
```diff
@@ -135,6 +131,24 @@
          </router-connector>
       </replacement>
    </supplement>
+   <supplement name="memcached-local">
+      <replacement placeholder="@@default-cache-container@@" attributeValue="local" />
+      <replacement placeholder="CONNECTORS">
+         <memcached-connector socket-binding="memcached" cache-container="local" cache="default"/>
```
Comment (mgencur) [inline server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml]: I thought memcached required "memcachedCache". Is this limitation fixed already?

File: server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml [hunk 2]
```diff
@@ -135,6 +131,24 @@
...
+      <replacement placeholder="CONNECTORS">
+         <memcached-connector socket-binding="memcached" cache-container="clustered" cache="default"/>
+      </replacement>
+   </supplement>
+   <supplement name="websocket-local">
+      <replacement placeholder="@@default-cache-container@@" attributeValue="local" />
+      <replacement placeholder="CONNECTORS">
+         <websocket-connector socket-binding="websocket" cache-container="local" />
+      </replacement>
+   </supplement>
```
Comment (mgencur) [inline server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml]: websocket-clustered is left out on purpose?

File: server/integration/testsuite/pom.xml [hunk 3]
```diff
@@ -359,7 +358,6 @@
         </groups.client.local.domain>
         <groups.client.clustered>
             org.infinispan.server.test.category.HotRodClustered,
-            org.infinispan.server.test.category.MemcachedClustered,
```
Comment (mgencur) [inline server/integration/testsuite/pom.xml]: You're removing it here but not removing it from suite.client.dist.domain.phase and suite.client.local.domain.phase so the  domain profile will be broken, IMO.

File: server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml (hunk 1 above)
Comment (tristantarrant) [inline server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml]: That is the default cache name indeed. But here I've overridden it.

File: server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml (hunk 2 above)
Comment (tristantarrant) [inline server/integration/endpoint/src/main/resources/subsystem-templates/infinispan-endpoint.xml]: Yes, that has no integration tests, and I think it is even less popular than memcached.

File: server/integration/testsuite/pom.xml (hunk 3 above)
Comment (tristantarrant) [inline server/integration/testsuite/pom.xml]: Right, will fix it.

**Comment by mgencur** [comment]
I can still see socket-bindings for memcached and rest when I build the server.

File: server/integration/feature-pack/feature-pack-build.xml [hunk 4]
```diff
@@ -97,7 +97,22 @@
...
+           subsystems="configuration/examples/subsystems-memcached-clustered.xml"
+           output-file="docs/examples/configs/clustered-memcached.xml"/>
+        <domain
+           template="configuration/domain/template.xml"
+           subsystems="configuration/examples/subsystems-memcached-domain.xml"
+           output-file="docs/examples/configs/domain-memcached.xml" />
+        <standalone
+           template="configuration/standalone/template.xml"
+           subsystems="configuration/examples/subsystems-websocket-local.xml"
+           output-file="docs/examples/configs/standalone-websocket-local.xml"/>
```
Comment (mgencur) [inline server/integration/feature-pack/feature-pack-build.xml]: This has the suffix -local while the memcached equivalent is only called standalone-memcached.xml

**Comment by tristantarrant** [comment]
I don't think there is a way to avoid socket-bindings, as they are defined in the global section of the template.
As for the REST connector not being in the default config, let's hear if anybody has opinions on this.

**Comment by tristantarrant** [comment]
Still need to fix a bunch of tests which expect the Memcached MBeans being present ...

**Comment by tristantarrant** [comment]
Finally fixed all REST ITs by making the rest helper non-static!!!

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/client/rest/AbstractRESTAsyncIT.java [hunk 5]
```diff
@@ -67,20 +64,20 @@ public void testPutOperation() throws Exception {
 
         long t1 = System.currentTimeMillis();
         for (int i = 0; i < NUM_OPERATIONS; i++) {
-            put(fullPathKey, initialXML, "application/octet-stream", HttpStatus.SC_OK, "performAsync", "false");
+            rest. put(fullPathKey, initialXML, "application/octet-stream", HttpStatus.SC_OK, "performAsync", "false");
```
Comment (mgencur) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/client/rest/AbstractRESTAsyncIT.java]: Is there a space? Does it work?

**Comment by wburns** [comment]
Pulling.. I will fix space

**Comment by wburns** [comment]
Integrated into master, thanks @tristantarrant !

**Pull request closed without merging**